src/crypto/internal/boring/aes.go

```diff
@@ -69,10 +76,34 @@ func (c *aesCipher) Encrypt(dst, src []byte) {
 	if len(dst) < aesBlockSize {
 		panic("crypto/aes: output not full block")
 	}
-	C._goboringcrypto_AES_encrypt(
-		(*C.uint8_t)(unsafe.Pointer(&src[0])),
-		(*C.uint8_t)(unsafe.Pointer(&dst[0])),
-		&c.enc)
+
+	if c.enc_ctx == nil {
+		c.enc_ctx = C.EVP_CIPHER_CTX_new()
+		if c.enc_ctx == nil {
+			panic("cipher: unable to create EVP cipher ctx")
+		}
+
+		k := (*C.uchar)(unsafe.Pointer(&c.key[0]))
+
+		var cipher *C.EVP_CIPHER
+		switch len(c.key) * 8 {
+		case 128:
+			cipher = C.EVP_aes_128_ecb()
+		case 192:
+			cipher = C.EVP_aes_192_ecb()
+		case 256:
+			cipher = C.EVP_aes_256_ecb()
+		default:
+			panic("invalid key size")
+		}
```

Comment:
If you move the switch to NewAESCipher you can actually report the wrong key length as error (return NULL). Also you can spare the second copy of the switch below.

Reply:
good refactor, done.